src/dss/logs.py

```diff
@@ -0,0 +1,76 @@
+from lightkube import ApiError, Client
+from lightkube.resources.apps_v1 import Deployment
+from lightkube.resources.core_v1 import Pod
+
+from dss.config import DSS_NAMESPACE, MLFLOW_DEPLOYMENT_NAME
+from dss.logger import setup_logger
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def get_logs(parts: str, name: str, lightkube_client: Client) -> None:
+    """
+    Retrieve logs from specified parts of the DSS application.
+
+    Args:
+        parts (str): Specify which part's logs to retrieve: 'notebooks' or 'mlflow'.
+        name (str): Name of the notebook or deployment to retrieve logs from.
+        lightkube_client (Client): The Kubernetes client.
+
+    Returns:
+        None
+    """
+    try:
+        if parts == "notebooks":
+            deployment = lightkube_client.get(Deployment, name=name, namespace=DSS_NAMESPACE)
+        elif parts == "mlflow":
+            deployment = lightkube_client.get(
+                Deployment, name=MLFLOW_DEPLOYMENT_NAME, namespace=DSS_NAMESPACE
+            )
+        else:
+            logger.error("Invalid value for 'parts'. Use 'notebooks' or 'mlflow'.")
+            return
+    except ApiError:
+        deployment = None
+
+    # The get command can return empty DeploymentList object with empty spec field
+    if deployment is None or deployment.spec is None:
+        if parts == "notebooks":
+            logger.error(f"No notebook found with the name '{name}'.")
+            logger.info(
+                "Please wait a moment or recreate the notebook with 'dss create-notebook'."
+            )
+        elif parts == "mlflow":
+            logger.error("MLflow is not initialized properly.")
+            logger.info("Please run 'dss initialize' first.")
+        return
+
+    selector_labels = deployment.spec.selector.matchLabels
+    pods = lightkube_client.list(Pod, namespace=DSS_NAMESPACE, labels=selector_labels)
+
+    if not pods:
+        if parts == "notebooks":
+            logger.error(f"No pods found for notebook - {name}.")
+            logger.info(
+                "Please wait a moment or recreate the notebook with 'dss create-notebook'."
+            )
+        elif parts == "mlflow":
+            logger.error("No pods found for MLflow.")
+            logger.info("Please wait a moment or run 'dss initialize' first.")
+        return
+
+    for pod in pods:
```

Comment:
Please correct me if I am wrong. I understand this will print the logs from the Notebook server and from each of the Jupyter notebooks that get created via `dss create-notebook` OR from the web app, right?

If that's the case, I think the output will be huge if users have a lot of notebooks. Is there a way to filter out a bit more the result? Say, by specifying the name of the actual notebook (not the notebook server)?

Reply:
When we will run 
```
dss create notebook-name
```
that command will create exactly one Deployment with one Pod and inside will be one container with jupyter-lab instance (lets call it lab because server is confusing). Inside the lab users will create notebooks but everything will stay in one pod in one container. There is no simple way to extract logs only for one notebook. From my practical experience you want to see the labs logs if there is some kernel problem or web app problem. I cannot even come up with some practical logs which would be specific only to the notebooks.